frontend/app/src/components/dashboard/edit-snapshot/EditBalancesSnapshotAssetPriceForm.vue

```diff
@@ -0,0 +1,313 @@
+<script setup lang="ts">
+import { toMessages } from '@/utils/validation';
+import { CURRENCY_USD } from '@/types/currencies';
+import { TaskType } from '@/types/task-type';
+import { ApiValidationError, type ValidationErrors } from '@/types/api/errors';
+import type { Validation } from '@vuelidate/core';
+import type { BigNumber } from '@rotki/common';
+import type { HistoricalPriceFormPayload } from '@/types/prices';
+import type { NewHistoryEventPayload } from '@/types/history/events';
+import type { ActionStatus } from '@/types/action';
+
+const props = withDefaults(
+  defineProps<{
+    timestamp: number;
+    asset: string;
+    disableAsset?: boolean;
+    v$: Validation;
+    nft?: boolean;
+  }>(),
+  {
+    disableAsset: false,
+    nft: false,
+  },
+);
+
+const emit = defineEmits<{
+  (e: 'update:asset', asset: string): void;
+}>();
+
+const { timestamp, asset, disableAsset } = toRefs(props);
+
+const assetModel = computed({
+  get() {
+    return get(asset);
+  },
+  set(asset: string) {
+    if (get(disableAsset))
+      return;
+
+    emit('update:asset', asset);
+  },
+});
```

Comment:
two questions here:

1. is there any reason to not use `defineModel` for asset
2. would it make sense to extract the logic around assetModel to a util function (I tried searching for something but I couldn't find anything existing.)? Also is there truly a case where it would update the value when asset is disabled?